### PR TITLE
Fixed - Download link in order email shows incorrect escaped link

### LINKED ISSUE #28014: Download link in order email shows incorrect escaped link
### Preconditions (*)
Magento 2.4-develop
Php 7.3

### Steps to reproduce (*)
1. Create a simple product with **File** customizable option;
Screenshot from 2020-04-29 16-13-29

2. Go to Storefront;
3. Add created before the product to Cart;
Screenshot from 2020-04-29 16-21-52
4. Finish the order and look at the mail

### Expected result (*)

1. The mail provides a readable  link to the download

### Actual result (*)

:heavy_multiplication_x:  The attribute value is encodes, so it shows the raw HTML instead of a formatted link.
screenshot_1

This was introduced by escaping the attribute value. A similar bug was encountered for the minicart. This is fixed by checking the type and then rendering it raw (see https://github.com/magento/magento2/pull/24237) . This should probably be done similarly for the emails.

This escaping happens here, but probably in more emails (credit, invoice etc): https://github.com/magento/magento2/blob/f442fa9a1f10cc566053e60ed6ae099a1664e61a/app/code/Magento/Sales/view/frontend/templates/email/items/order/default.phtml#L24
When the type is file, it should not escape.

### PR DESCRIPTION
**Pull request by niravkrish**
### Description (*)

- Fixed issue with escape html for option type file for order, invoice, shipment as well as creditmemo email template.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#28014: Download link in order email shows incorrect escaped link

### Manual testing scenarios (*)

1. Tested in magento 2.3.3 instance for order, invoice,shipment as well as creditmemo email.

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Sales/view/frontend/templates/email/items/creditmemo/default.phtml [hunk 1]
```diff
@@ -16,7 +16,14 @@
                 <?php foreach ($block->getItemOptions() as $option) : ?>
                     <dt><strong><em><?= $block->escapeHtml($option['label']) ?></em></strong></dt>
                     <dd>
-                        <?= /* @noEscape */  nl2br($block->escapeHtml($option['value'])) ?>
+                        <?php
+                        if ($option['option_type'] == 'file') {
```
Comment (edenduong) [inline app/code/Magento/Sales/view/frontend/templates/email/items/creditmemo/default.phtml]: Should be \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_FILE

File: app/code/Magento/Sales/view/frontend/templates/email/items/invoice/default.phtml [hunk 2]
```diff
@@ -15,7 +15,14 @@
                 <?php foreach ($block->getItemOptions() as $option) : ?>
                     <dt><strong><em><?= $block->escapeHtml($option['label']) ?></em></strong></dt>
                     <dd>
-                        <?= /* @noEscape */  nl2br($block->escapeHtml($option['value'])) ?>
+                        <?php
+                        if ($option['option_type'] == 'file') {
```
Comment (edenduong) [inline app/code/Magento/Sales/view/frontend/templates/email/items/invoice/default.phtml]: Should be \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_FILE

File: app/code/Magento/Sales/view/frontend/templates/email/items/order/default.phtml [hunk 3]
```diff
@@ -21,7 +21,14 @@ $_order = $_item->getOrder();
             <?php foreach ($block->getItemOptions() as $option) : ?>
                 <dt><strong><em><?= $block->escapeHtml($option['label']) ?></em></strong></dt>
                 <dd>
-                    <?= /* @noEscape */  nl2br($block->escapeHtml($option['value'])) ?>
+                    <?php
+                    if ($option['option_type'] == 'file') {
```
Comment (edenduong) [inline app/code/Magento/Sales/view/frontend/templates/email/items/order/default.phtml:25]: Should be \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_FILE

File: app/code/Magento/Sales/view/frontend/templates/email/items/shipment/default.phtml [hunk 4]
```diff
@@ -15,7 +15,14 @@ $_item = $block->getItem() ?>
                 <?php foreach ($block->getItemOptions() as $option) : ?>
                     <dt><strong><em><?= $block->escapeHtml($option['label']) ?></em></strong></dt>
                     <dd>
-                        <?= /* @noEscape */  nl2br($option['value']) ?>
+                        <?php
+                        if ($option['option_type'] == 'file') {
```
Comment (edenduong) [inline app/code/Magento/Sales/view/frontend/templates/email/items/shipment/default.phtml]: Should be \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_FILE

**Review by edenduong — CHANGES_REQUESTED** [review]
Could you please fix static test and replace 'file' to constant \Magento\Catalog\Api\Data\ProductCustomOptionInterface::OPTION_TYPE_FILE

File: app/code/Magento/Sales/view/frontend/templates/email/items/order/default.phtml (hunk 3 above)
Comment (niravkrish) [reply app/code/Magento/Sales/view/frontend/templates/email/items/order/default.phtml:25]: I have applied changes as per suggestion as well as I have tried and fixed some phpcs errors.

File: app/code/Magento/Sales/view/frontend/templates/email/items/shipment/default.phtml (hunk 4 above)
Comment (niravkrish) [reply app/code/Magento/Sales/view/frontend/templates/email/items/shipment/default.phtml]: I have applied changes as per suggestion as well as I have tried and fixed some phpcs errors.

**Review by edenduong — DISMISSED** [review]
Thank you for your contribution !

**Comment by magento-engcom-team** [comment]
Hi @edenduong, thank you for the review.
[ENGCOM-7502](https://jira.corp.magento.com/browse/ENGCOM-7502) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Create a simple product with **File** customizable option;
Screenshot from 2020-04-29 16-13-29

2. Go to Storefront;
3. Add created before the product to Cart;
Screenshot from 2020-04-29 16-21-52
4. Finish the order and look at the mail

**Before:** :heavy_multiplication_x: shows the raw HTML instead of a formatted link

screenshot_1

**After:** :heavy_check_mark: The mail provides a readable link to the download
screenshot_3

Were also tested for order, invoice, shipment, credit memo email, and everything works as expected.

**Comment by engcom-Echo** [comment]
@magento run Functional Tests B2B

**Comment by slavvka** [comment]
@magento run all tests

**Review by VladimirZaets — CHANGES_REQUESTED** [review]
Hi @niravkrish. Thanks for collaboration. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. In this case, I think it can be covered by an integration test. Can you do it?

**Comment by niravkrish** [comment]
> Hi @niravkrish. Thanks for collaboration. Due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all code must be covered by tests. In this case, I think it can be covered by an integration test. Can you do it?

I am not aware about test cases and I haven't developed. so I can not able to do it.

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

**Comment by engcom-Echo** [comment]
Hi @niravkrish. Please provide write access for maintainers.

**Comment by niravkrish** [comment]
@engcom-Echo 
I have tried to give an access but it is not working it remains the same.

**Comment by engcom-Echo** [comment]
Hi @niravkrish, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**